test/actor_test.py

```diff
@@ -1733,13 +1732,38 @@ def fork(queue, key, num_items):
             x = None
             for item in range(num_items):
                 x = queue.enqueue.remote(key, item)
+            # Wait for the last task to complete.
             return ray.get(x)
 
         # Fork num_iters times.
         num_forks = 10
         num_items_per_fork = 100
         ray.get([fork.remote(queue, i, num_items_per_fork) for i in
                  range(num_forks)])
+        # Test that after the forks return, we have guarantee per-handle
+        # execution ordering.
+        items = ray.get(queue.read.remote())
+        for i in range(num_forks):
+            filtered_items = [item[1] for item in items if item[0] == i]
+            self.assertEqual(filtered_items, list(range(num_items_per_fork)))
+
+    def testJoinConsistency(self):
+        queue = self.setup_queue_actor()
+
+        # A fork task that does not wait for any of the methods that it submits
+        # on an actor to complete.
+        @ray.remote
+        def fork(queue, key, num_items):
+            for item in range(num_items):
+                queue.enqueue.remote(key, item)
+
+        # Fork num_iters times.
+        num_forks = 10
+        num_items_per_fork = 100
+        ray.get([fork.remote(queue, i, num_items_per_fork) for i in
+                 range(num_forks)])
+        # Test that after the forks return, we have guarantee per-handle
+        # execution ordering.
```

Comment:
The way that this test would fail before this PR is that the `queue.read.remote` below would execute before some of the `queue.enqueue.remote` tasks, right?

Reply:
Yup! I also confirmed that the test fails without this PR. :)